website/docs/cdktf/concepts/providers-and-resources.html.md

```diff
@@ -264,6 +264,30 @@ export class HelloTerra extends TerraformStack {
 
 The [examples page](/docs/cdktf/examples.html) contains multiple example projects for every supported programming language.
 
+### Using References
+
+When working with resources created from providers you can use properties on the instances from the generated bindings to reference in other resources. This allows us to track logical dependencies and access computed values.
```

Comment:
Okay I have a couple questions before I rework this a bit :-) 

- Can you give a real-world example of when this is useful (beyond the example below)?
- Can we give an explanation of what's happening in the example below?
- How can folks find the properties - should they just check the provider documentation for available ones?

Reply:
> Can you give a real-world example of when this is useful (beyond the example below)?

Every time they would normally write a reference in HCL, so if there is a value or logical dependency between two items. E.g. a new Role being created and being referenced by ARN in some assignment. A logical dependency would be in the Kubernetes space, resource there live in namespaces, so the namespace needs to be there before the resource. If I don't use a reference TF might want to provision resources before the namespace is there. When using a reference the order is clear, a dependency has been established

> Can we give an explanation of what's happening in the example below?

In the example we create a namespace and us its name as the namespace property in the Deployment. This creates an implicit dependency between the two and the value of the namespace's name is used in the Deployment. If we used generateName that generates a random name in kubernetes we could still reference the name and it's computed value would be used through the reference. I hope this sort of makes sense 🙈 


> How can folks find the properties - should they just check the provider documentation for available ones?

Yeah the provider documentation is the best place to look at

Reply:
Ah I think I see your confusion now. Metadata is a resource attribute / block, it has nothing to do with references. Using a property on a resource (e.g., `myResource.whateverProperty`) inside of the configuration of another construct is what creates the reference